qiskit_nature/operators/second_quantization/vibrational_op.py

```diff
@@ -0,0 +1,338 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A Vibration operator."""
+
+from functools import reduce
+from typing import List, Tuple, Union, Optional
+
+import re
+
+import numpy as np
+
+from qiskit_nature import QiskitNatureError
+
+from .vibrational_op_utils.vibrational_labels_validator import _validate_vibrational_labels
+from .vibrational_op_utils.vibrational_dense_label_converter import _convert_to_dense_labels
+
+from .second_quantized_op import SecondQuantizedOp
+
+
+class VibrationalOp(SecondQuantizedOp):
+    """Vibration type operators.
+
+    **Label**
+
+    :class:`VibrationalOp` supports two kinds of labels: sparse and dense ones.
+
+    1. Sparse Labels:
+
+    Allowed characters for primitives of labels are `+` and `-`.
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Label
+          - Mathematical Representation
+          - Meaning
+        * - `+`
+          - :math:`S_+`
+          - Raising operator
+        * - `-`
+          - :math:`S_-`
+          - Lowering operator
+
+    :class:`VibrationalOp` accepts the notation that encodes raising (`+`) and lowering (`-`)
+    operators together with indices of modes and modals that they act on, e.g.
+    `+_{mode_index}*{modal_index}`. Each modal can be excited at most once.
+
+    For more information on how the sparse label notation works, please refer to the more extensive
+    examples in the :class:`SpinOp` class.
+
+    2. Dense Labels:
+
+    Internally, a :class:`VibrationalOp` is stored as a dense label in an identical notation to that
+    of the :class:`FermionicOp`. Thus, each modal is mapped onto a position of the string and can be
+    one of the following characters: `I`, `+`, `-`, `N` and `E`. Here, `N` and `E` are shorthands
```

Comment:
```suggestion
    one of the following characters: `I`, `+`, `-`, `N` and `E`. Here, `N` and `E` are shorthand
```
Very minor comment. To me the plural form reads oddly here - saying they are shorthand for ie short for in seems fine. It also means its not needed in the custom dictionary so I guess its not common in the plural